src/helper/ssg/ssg.ts

```diff
@@ -62,17 +67,22 @@ const parseResponseContent = async (response: Response): Promise<string | ArrayB
   }
 }
 
-const determineExtension = (mimeType: string): string => {
-  switch (mimeType) {
-    case 'text/html':
-      return 'html'
-    case 'text/xml':
-    case 'application/xml':
-      return 'xml'
-    default: {
-      return getExtension(mimeType) || 'html'
-    }
+const defaultExtensionMap: Record<string, string> = {
+  'text/html': 'html',
+  'text/xml': 'xml',
+  'application/xml': 'xml',
+  'application/yaml': 'yaml',
+}
+
+const determineExtension = (
+  mimeType: string,
+  userExtensionMap?: Record<string, string>
+): string => {
+  const extensionMap = { ...defaultExtensionMap, ...userExtensionMap }
```

Comment:
I think that we can export the `defaultExtensionMap` and the user can use it and add what they want to add to it. Or the user could not use the `defaultExtensionMap`.

```ts
import { toSSG, defaultExtensionMap } from 'hono/ssg'
import { Hono } from 'hono'

//...

toSSG(app, fs, {
  extensionMap: { ...defaultExtensionMap, ...{ 'application/foo': 'foo' } }
})
```

This idea from Vitest. We can add the `exclude` paths with `defaultExclude`:

```ts
import { defineConfig, defaultExclude } from 'vitest/config'

export default defineConfig({
  test: {
    // defaultExclude is defined at:
    // https://github.com/vitest-dev/vitest/blob/fee7d8be9d6e6f710270600ae91fa35d861b7075/packages/vitest/src/defaults.ts#L6
    exclude: [...defaultExclude, 'abc', 'def']
  }
})
```

How do you think about this?

Reply:
The nice thing about this is that users can add values and rewrite the default values.

Reply:
Yes, let's make the default value a user-accessible method. Fixed.